paws_collector.js

```diff
@@ -549,12 +549,30 @@ class PawsCollector extends AlAwsCollector {
         Promise.all(promises).then((res) => {
             return callback(null, privCollectorState, nextInvocationTimeout);
         }).catch((err) => {
-            collector.reportErrorToIngestApi(err, () => {
-                return callback(err);
+            let error = collector.parseError(err);
+            collector.reportErrorToIngestApi(error, () => {
+                return callback(error);
             });
         });
     }
 
+    parseError(error) {
```

Comment:
it sounds like this function belongs to `al-collector-js` because same function might be helpful for Azure collectors as well.

Reply:
Let's maybe move it to `https://github.com/alertlogic/al-collector-js/blob/npm/al_util.js` or create a new `al_error.js` module for all error handling functions.

Reply:
Or IMHO a better idea is to return error objects with `errorCode` property from `al-aws-collector-js`. For example change this https://github.com/alertlogic/al-aws-collector-js/blob/master/al_aws_collector.js#L681
to return 
```
{
  errorCode: AWSC0018,
  errorMesasges: "AWSC0018 failed to send the logmsgs : ${exception.message}"
  httpErrorCode: 404
}
```

And amend `al-aws-collector-js` lib to extract HTTP error codes and pass them as separate field.

Reply:
move the code in al-aws-collector and also return error object from it. Reference [pr](https://github.com/alertlogic/al-aws-collector-js/pull/88)